### PR TITLE
feat: connect with tracking devices to retrieve GPS data 

### PR DESCRIPTION
**Pull request by brunobesson**
This PR is related to [c2c_tracking](https://github.com/c2corg/c2c_tracking) new backend module.

The idea is to allow users to connect their c2c account with devices trackers such as Suunto, or Strava. Currently Garmin, Decathlon, Strava and Suunto are supported. Polar support is to be added soon.

Once a connection is established, activities are retrieved (and webhooks configured to retrieve later activities). Globaly, the last 30 activities are retained. Then, when uploading a GPX track to e.g. a route or outing, a modal dialog allows to directly choose an activity and retrieve the corresponding geometry. It is still possible to choose a local file, of course.

For now, because we still need more tests & all before requesting production application, this feature is accesible to modrators only. This should be removed ASAP.

Some screenshots: 

image
image
image

Technically I also replaced `date-fns` with `dayjs`, which I wanted to do for some time for other reasons, as it simplified the way we can show local date time (`date-fns` converts the date in the local datetime, wihle `dayjs` ignore the timezone).

I also cleaned up some code related to API services.

### CODE REVIEW

**Comment by brunobesson** [comment]
Note: test it on www.demov6.camptocamp.org

**Review by lbesson — COMMENTED** [review]
Shouldn't we also be able to completely disable the feature (so that if this is merged, we can release ui updates before the server side is ready?)

File: src/components/map/GpsFileUpload.vue [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      }.bind(this);
+
+      const file = event.target.files[0];
+      if (file.type === 'application/vnd.ant.fit' || file.name.toLowerCase().endsWith('.fit')) {
+        reader.readAsArrayBuffer(file);
+      } else {
+        reader.readAsText(file);
+      }
+
+      // empty the input, because if user wan't to upload same trace
```
[the remark this answers is not in the document]
Comment (brunobesson) [reply src/components/map/GpsFileUpload.vue:32]: indeed, although copied and pasted for my defense 😇

**Comment by brunobesson** [comment]
> Shouldn't we also be able to completely disable the feature (so that if this is merged, we can release ui updates before the server side is ready?)

I understand what you mean. But I do not plan to merge and deploy this UI feature until backend is effectively deployed. If we do so, there is no need to do as you suggest?

**Comment by lbesson** [comment]
the menu icon seems to be completely dark, while other icons are dark grey?

**Comment by lbesson** [comment]
As for the list of activities, probably you can make the date shorter (only day, not hour), with complete time on hover?
You usually don't have more than one activity per day

**Comment by lbesson** [comment]
some more (not vital) feedback: the UI says "select the activity" but where to click is not explicit, you have the download icon (I guess it doesn't select), the activity link (same), etc...

**Comment by brunobesson** [comment]
> some more (not vital) feedback: the UI says "select the activity" but where to click is not explicit, you have the download icon (I guess it doesn't select), the activity link (same), etc...

Yeah, I am not totally convinced with curent result either. Would you have a proposal?

**Comment by brunobesson** [comment]
> the menu icon seems to be completely dark, while other icons are dark grey?

Only when hovered (same for others), unless I missed something?

**Comment by lbesson** [comment]
Some activities should be automatically excluded (e.g. swimming, or at least detect beforehand if geographical data is attached?)

**Comment by brunobesson** [comment]
> Some activities should be automatically excluded (e.g. swimming, or at least detect beforehand if geographical data is attached?)

There is a **huge** amount of activities (e.g. for Decathlon, less for others but it is consistent anyway).
I chose not to filter because sometimes you moght have a sport type which you upate later, so you might loose some stuff.

The geographical stuff is a more interesting idea IMHO. For now, I have decided to retrieve it when needed (unless for Garmin and Polar, where we need o retrieve it ASAP), but maybe we could choose to change and retrieve geographical data frmo the begining (with the problem of handling errors...)

**Comment by brunobesson** [comment]
Thanks for your review @lbesson 
I'll merge the PR for now to enable feedback from the community, we can enhance the UI later.

**Pull request merged**